Added Marley Script

jpkneller opened the pull request:
Script to create a tarfile containing MARLEY config files from a particular SN model

jpkneller committed:
Add files via upload

Script to create a tarfile containing MARLEY config files from a particular SN model

JostMigenda commented on python/snewpy/scripts/snewpy_to_marley.py:
```diff
@@ -0,0 +1,95 @@
+times    = snmodel.time
+energies = np.linspace(0,50,501)<<u.MeV
+
+d = (10*u.kpc).to('cm').value # distance to SN
+    
+mp_nmo = MixingParameters()
+xform = AdiabaticMSW(mp_nmo)
+
+flux = snmodel.get_flux(t=times, E=energies,  distance=d, flavor_xform=xform)
```
```suggestion
d = 10 * u.kpc  # distance to SN
    
mp_nmo = MixingParameters()
xform = AdiabaticMSW(mp_nmo)

flux = snmodel.get_flux(t=times, E=energies,  distance=d, flavor_xform=xform)
```
The `get_flux` function is able to handle Astropy quantities, so there’s no need to convert to a numerical value. (Additionally, if the `distance` is given as a unit-less number, `get_flux` will [assume that it is in kpc](https://github.com/SNEWS2/snewpy/blob/17f14724e679261f985bc503531b630030ee4172/python/snewpy/models/base.py#L182), so converting to a cm value will give a wrong result.

JostMigenda commented on python/snewpy/scripts/snewpy_to_marley.py:
```diff
@@ -0,0 +1,95 @@
+
+import logging
+import os
+import re
+import tarfile
+import io
+import json
+
+import numpy as np
+import pandas as pd
+from astropy import units as u
+from warnings import warn
```
```suggestion
import re
import tarfile
import io
import json

import numpy as np
from astropy import units as u
```
Remove unused imports.

JostMigenda commented on python/snewpy/scripts/snewpy_to_marley.py:
```diff
@@ -0,0 +1,95 @@
+def save_as_marley(fluence_object, output_filename, marley_additional_inputs=None):
+    """Save the contents of a fluence object in MARLEY format.
+
+       For each time bin 6 Marley config files are genreated and placed into a tarfile.
```
```suggestion
       For each time bin, 6 Marley config files are generated and placed into a tarfile.
```

JostMigenda commented on python/snewpy/scripts/snewpy_to_marley.py:
```diff
@@ -0,0 +1,95 @@
+       ----------
+       fluence_object : flux container
+           Input file containing neutrino flux information from supernova model.
+       marley_additional_inputs : dict
```
It might be nice to rename this to `additional_inputs`, to be a bit less unwieldy?

JostMigenda commented on python/snewpy/scripts/snewpy_to_marley.py:
```diff
@@ -0,0 +1,95 @@
+    	   Path of tarfile with the Marley config files.
+       """
+
+    times=fluence.time
```
Should this be `fluence_object.time`?

(Actually, it might be nice to rename that function argument to just `fluence`. The `_object` suffix just takes up space but doesn’t add any information—everything is an object in Python.)

JostMigenda reviewed (changes requested):
I’ve noticed one subtle issue which could give wrong results (by some orders of magnitude), plus a few other bits of feedback. See the individual comments for details.

By the way, have we heard back from the MARLEY user(s) who wanted to test this? It would be good to get their feedback and confirm it works before we merge this.

jpkneller committed:
Update python/snewpy/scripts/snewpy_to_marley.py

Co-authored-by: Jost Migenda <jost.migenda@kcl.ac.uk>

jpkneller committed:
Update python/snewpy/scripts/snewpy_to_marley.py

Co-authored-by: Jost Migenda <jost.migenda@kcl.ac.uk>

jpkneller committed:
Update python/snewpy/scripts/snewpy_to_marley.py

Co-authored-by: Jost Migenda <jost.migenda@kcl.ac.uk>

jpkneller committed:
Update python/snewpy/scripts/snewpy_to_marley.py

Co-authored-by: Jost Migenda <jost.migenda@kcl.ac.uk>

jpkneller committed:
Update snewpy_to_marley.py

Implemented Jost's suggestions

JostMigenda reviewed (approved):
All my comments are resolved.

I’d still prefer to wait for feedback from actual MARLEY users; but if they continue to be unresponsive for the next few weeks, at some point we can just go ahead and merge this as is.

jpkneller committed:
Add files via upload

Updated script after getting feedback from Yun-Tse Tsai. The script now inserts into the MARLEY config files all the info MARLEY needs to run

jpkneller commented:
I implemented the feedback I got from Yun-Tse Tsai and now the script inserts all the pieces into the config files that MARLEY needs.
